OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs

```diff
@@ -134,14 +141,25 @@ private async Task<IEnumerable<RatingDto>> GetNewRatings(DateTimeOffset lastLaun
     {
         logger.LogInformation("The average ratings calculation started.");
 
+        var filter = new OperationWithObjectFilter() { OperationType = OperationWithObjectOperationType.RecalculateAverageRating };
+        var operations = await operationWithObjectService.GetAll(filter);
+        var workshopsIdsFromOperations = operations.Select(x => (Guid)x.EntityId);
+
         var workshopsIds = GetUniqueWorkshopIds(ratings);
+        workshopsIds = workshopsIds.Union(workshopsIdsFromOperations);
+
         var providersIds = GetUniqueProviderIdsByWorkshopIdsAsync(workshopsIds);
 
         Dictionary<Guid, Tuple<float, int>> averageRatings = new Dictionary<Guid, Tuple<float, int>>();
 
         averageRatings.AddRange(CalculateWorkshopsRatingsByEntityIdsAsync(workshopsIds));
         averageRatings.AddRange(CalculateProvidersRatingsByProvidersIdsAsync(providersIds));
 
+        foreach (var operation in operations)
```

Comment:
maybe run this in transaction with what ever logic it runs in?

or it does not matter if we fail to delete a rating operation and just try it later some time?

Reply:
The last one ))))